spring-data-neo4j-rx/src/main/java/org/neo4j/springframework/data/core/ReactiveNeo4jTemplate.java

```diff
@@ -307,15 +310,21 @@ public ReactiveNeo4jTemplate(ReactiveNeo4jClient neo4jClient, Neo4jMappingContex
 
 				Collection<RelationshipDescription> relationships = neo4jPersistentEntity.getRelationships();
 
-				Class<?> associationTargetType = inverse.getAssociationTargetType();
-
-				Neo4jPersistentEntity<?> targetNodeDescription = (Neo4jPersistentEntity<?>) neo4jMappingContext
-					.getRequiredNodeDescription(associationTargetType);
-
 				RelationshipDescription relationship = relationships.stream()
 					.filter(r -> r.getFieldName().equals(inverse.getName()))
 					.findFirst().get();
 
+				Class<?> associationTargetType;
+				// if we have a relationship with properties, the targetNodeType is the Map Key
+				if (relationship.hasRelationshipProperties()) {
```

Comment:
Could be a conditional/ternary operation to align with the style below.

Reply:
See latest two commits